### ISSUE TITLE
Question: Upload backups to EMC ECS ?

### ISSUE THREAD

**Issue by akamalov** [pivotal-cf/bbr-pcf-pipeline-tasks#14]
Does``` bbr-pcf-pipeline``` support uploading backups to EMC ECS ?

**Comment by akamalov** [comment]
I am using EMC ECS and I am repeatedly getting errors trying to backup to ECS.

Trying to push Ops Manager backup zip file:
```
om-backup-artifact
3.42 MB / 3.42 MB [===================================] 100.00 % 18.79 KB/s 3m6serror running command: RequestError: send request failed
caused by: Put https:/pcf-cflab02-cg-backups/installation.zip: Proxy Error
```
If you take a look at output above, it's missing EMC ECS endpoint (i.e., ``` s3ecs.XXXXX.localnet:9020")``` and referencing an empty string right after ```https```, followed with a bucket name (```pcf-cflab02-cg-backups```).

Trying to push ERT backup:

```
ert-backup-bucket
5.00 MB / 113.51 MB [=>--------------------------------] 4.40 % 24.98 KB/s 3m24serror running command: RequestError: send request failed
caused by: Post https:/pcf-cflab02-cg-backups/ert-backup.tar?uploads=: Proxy Error
```
In both cases BBR pipeline is not picking up ```((storage-endpoint))``` variable.

Snippent from pipeline.yml:

```

- name: om-backup-artifact
  type: s3
  source:
    bucket: ((backup-artifact-bucket))
    region_name: ((storage-region))
    endpoint: ((storage-endpoint))
    access_key_id: ((storage-access-key-id))
    secret_access_key: ((storage-secret-access-key))
    versioned_file: installation.zip
    use_v2_signing: ((storage-use-v2-signing))

```

Snippet from secrets.yml file:

```
skip-ssl-validation: true
pivnet-api-token: "XXXXXXXXX"
bbr-version: 1.2.3
opsman-url: https://opsmanager.cflab02-cg.XXXXXXX.localnet
opsman-username: OpsMan
opsman-password: "XXXXXXXXX"
backup-artifact-bucket: pcf-cflab02-cg-backups
storage-region: ""
storage-endpoint: "http://s3ecs.XXXXX.localnet:9020"
storage-access-key-id: pcf-nonprod-01
storage-use-v2-signing: true
storage-secret-access-key: "XXXXXXXXXXXXXXXX"
concourse-worker-tag: BBR
private: true
disable_ssl: true
```

Does BBR pcf-pipeline support non-HTTPS uploads to ECS ? I am getting error 405.

Any ideas ?

Thank so much.

**Comment by akamalov** [comment]
Answering my own question: yes it does. You have to do couple of things:

1. Make sure you have enabled ```versioning``` on your bucket. This is a must
2. If you're using proxy in your environment, make sure to list your S3 in FQDN format in your ```/etc/environment``` file
3. As it currently stands, Concourse mangles ```no_proxy``` statement.  This is a bug. Concourse folks are envisioning a fix for it in 3.15.x release. As a workaround, remove all proxy-related entries in your worker environment file and source your ```/etc/environment``` file for your proxy entries. Consequently, in your SystemD file, pass proxy settings such as this ```--http-proxy=${http_proxy} --https-proxy=${https_proxy} --no-proxy=${no_proxy}``` and so forth. This should take are of it.

Cheers,

Alex
